app/controllers/application_controller.rb

```diff
@@ -3,5 +3,9 @@
 class ApplicationController < ActionController::Base
   # Only allow modern browsers supporting webp images, web push, badges, import maps, CSS nesting, and CSS :has.
   allow_browser versions: :modern
+
+  def authenticate_admin!
+    redirect_to root_path unless current_user&.is_admin?
+  end
```

Comment:
I put this on the `ApplicationController` so that other controllers can use it without duplicating this method all over the place.